src/sympc/tensor/replicatedshare_tensor.py

```diff
@@ -209,6 +252,158 @@ def ne(self, y):
 
         """
 
+    @staticmethod
+    def reconstruct_semi_honest(
+        share_ptrs: List["ReplicatedSharedTensor"],
+        get_shares: bool = False,
+    ) -> Union[torch.Tensor, List[torch.Tensor]]:
+        """Reconstruct value from shares.
+
+        Args:
+            share_ptrs (List[RSTensorPointers]): List of RSTensor pointers.
+            get_shares (bool): Retrieve only shares.
+
+        Returns:
+            reconstructed_value (torch.Tensor): Reconstructed value.
+        """
+        shares1 = share_ptrs[0].get_shares()[0].get()
+        shares2 = share_ptrs[1].get_shares().get()
+
+        shares = []
+
+        if isinstance(shares2, torch.Tensor):
+
+            shares = [shares1] + shares2
+
+        else:
+
+            shares = [shares1] + shares2
+
+        if get_shares:
+            return shares
+
+        return sum(shares)
+
+    @staticmethod
+    def reconstruct_malicious(
+        share_ptrs: List["ReplicatedSharedTensor"],
+        get_shares: bool = False,
+    ) -> Union[torch.Tensor, List[torch.Tensor]]:
+        """Reconstruct value from shares.
+
+        Args:
+            share_ptrs (List[RSTensorPointers]): List of RSTensor pointers.
+            get_shares (bool): Retrieve only shares.
+
+        Returns:
+            reconstructed_value (torch.Tensor): Reconstructed value.
+
+        Raises:
+            ValueError: When parties share values are not equal.
+        """
+        nparties = len(share_ptrs)
+
+        # Get shares from all parties
+
+        all_shares = []
+        for party_rank in range(nparties):
+
+            shares1 = share_ptrs[(party_rank)].get_shares()[0].get()
+            shares2 = share_ptrs[(party_rank + 1) % (nparties)].get_shares().get()
+
+            shares = []
+
+            if isinstance(shares2, torch.Tensor):
+                shares = [shares1] + [shares2]
+
+            else:
+                shares = [shares1] + shares2
+
+            if get_shares:
+                return shares
+
+            all_shares.append(shares)
+
+        # reconstruct shares from all parties and verify
+        value = None
```

Comment:
MAYBE: 
```
ref_shares, *to_check_shares = all_shares
ref_value = sum(ref_shares)
for shares in to_check_shares:
    if ref_value != sum(shares):
       raise...
```

Reply:
I had completely refactored the code because there was some weird behaviour calling get each time in the look. The new code is concise I guess.